make preview and customize char sections collapsible on portrait mode

Gaurav0 opened the pull request:
... for tablets and retina phones

Gaurav0 commented on sources/chargen.css:
```diff
@@ -258,6 +258,28 @@ summary {
+  }
+
+  #preview:has(+#controls>details:not([open])) {
+    margin-top: calc(170px + 12rex);
```
In theory, 1 rem is the width of the letter M in the font, and 1 rex is the width of the letter x in the font. The "r" for "root" indicates that we are talking about the font on the body, not the font on the element.

In practice, on Chrome and Safari, 16 px = 1 rem = 2 rex.

On retina displays, rem and rex units are respected while px units are multiplied by the dip (device/density independent pixel) size, usually 2.

So I'm mixing these units because on those retina displays, 12 rex is going to be 192 px while on regular displays they will be 96 px. This offsets that the body font size (set to 16px now) will be 16px on regular displays but 32px on retina displays.

Anyway, hope that makes sense. I suggest you review by looking at retina phones and tablets using the device picker in Chrome devtools.

jrconway3 reviewed (approved):
Apologies for the delay, but let's take a look at this...


The biggest issue I'm seeing is inconsistent text:
![image](https://github.com/user-attachments/assets/f99968f7-f546-4a74-86fc-c14b3eae67af)

The text size above of the "Preview" animation is barely readable, but after this, its GIGANTIC. Why is the text sizes so inconsistent?


One confusion here for me is that this view doesn't say anything about it being retina, even though it is an iOS view. However, when I first toggle to this view, it tells me:
980x2120. Not exactly sure what that means, and it doesn't sound like retina because iPhone 12 Pro says its 390 x 844...  IIRC, I thought retina was x2.


iPhone 12 Pro's landscape shows Preview being cut off:
![image](https://github.com/user-attachments/assets/27528a3a-1443-4cec-a156-7515e15526a2)

Both views of iPad Air are cutting of Preview:
![image](https://github.com/user-attachments/assets/ce3cc640-7c1e-4956-ad8c-94b71242970f)

Pretty sure iPad Air was working before.

iPad Pro looks fine, though

jrconway3 reviewed (changes requested):
I accidentally marked as approved rather than Request Changes. Please see prior comment.

jrconway3 commented:
Here's what it looks like on my Moto G Power:
![Screenshot_20250201-133507](https://github.com/user-attachments/assets/fe06c815-afea-41d8-a1c3-085d3b34d734)

And landscape:
![Screenshot_20250201-133546](https://github.com/user-attachments/assets/d73a4ac9-9aa0-45d0-b1c0-35600ade5532)


So the portrait view has the same issue as using chromium where the font sizes are inconsistent. However, landscape looks fine.

jrconway3 reviewed (changes requested):
This certainly looks a lot better, but now maybe its a bit TOO small...?
![image](https://github.com/user-attachments/assets/f2f54d02-93aa-4b88-92cf-0c0243857c78)


Originally some text was this small, and some text was ridiculously large. The large text is no longer happening, and it looks more consistent, so that's an improvement.

This looks very consistent now:
![image](https://github.com/user-attachments/assets/ae5bb942-460e-45b2-b8c5-245b295508e1)


Its still small, but it might look better on an actual mobile device.

Its still weirdly inconsistent in my Moto G Power... perhaps that's just because its cached?

jrconway3 commented:
Here's what it looks like on my Moto G Power:
![Screenshot_20250201-135250](https://github.com/user-attachments/assets/98052baf-a062-46ff-b0a5-99dee4493066)


I doubt this is something very easy to fix, it might even be cached or something. I would just recommend making the fonts slightly bigger than we can call of a day on this one

jrconway3 reviewed (changes requested):
When I load the view by default, I get this:
![image](https://github.com/user-attachments/assets/f6d37fe2-66e8-4848-9e91-3b031b9bc416)

Its not zooming in by default anymore for some weird reason. It zooms properly if I double tab, but not by default. This happens on mobile, too. It wasn't doing this before, so its... really weird.


This is inconsistent, with small text on some areas:
![image](https://github.com/user-attachments/assets/0412c7a8-b858-4aef-80c9-82266acc5dae)


However, when I CLICK on it, it restores to the correct size of the other links:
![image](https://github.com/user-attachments/assets/d02c28e7-d10d-4b74-afc2-af5b8f08ad87)

Gaurav0 commented:
@jrconway3 I ended up doing a refactor of the css.

- Removed the header and made everything use a display: grid
- Reordered some html
- Reverted the viewport change and the base font size change
- Created four separate sections to apply css
  - \> 1280px width
  - < 1280px width
  - < 1280 px width and <= 1200 px height
  - < 1280 px width and > 1200 px height

The css is much simpler now, and I'm also no longer seeing font sizing bugs in the devtools phone previews I was seeing before.

Hopefully it will be easier to fix any bugs you find now.

jrconway3 reviewed (changes requested):
I want to say "maybe this is just some sort of fluke on my end", but its happening on both my phone and in my Brave browser on my desktop:

![Screenshot_20250202-094114](https://github.com/user-attachments/assets/e2606f98-a3c3-4d6c-9b33-72c79e8b3eb0)
![Screenshot_20250202-094124](https://github.com/user-attachments/assets/c962d40a-455d-421d-b5a5-50fbbaa01e34)

Literally the exact same issue... the first two sections of the filters are significantly smaller.

jrconway3 commented:
Windows version (Brave):
![image](https://github.com/user-attachments/assets/53782d47-a91d-40fb-bfde-4bdc04fb12ad)


Even Edge:
![image](https://github.com/user-attachments/assets/7f869fce-ce92-4acb-b2fb-cf799c33978b)

Gaurav0 commented:
@jrconway3 For reference:

https://stackoverflow.com/questions/44378664/rem-based-layouts-zooming-on-chrome-is-inconsistent-px-vs-rem

I am thinking of specifying all font-size in pt.

Gaurav0 commented:
@jrconway3 Updated again. Fonts might be too small on some retina displays now, but at least it isn't all buggy.

jrconway3 reviewed (approved):
Everything looks fine now. I checked earlier and something still looked off, and I started leaving a comment, but apparently I forgot to post it. However, I'm not seeing anything now.
